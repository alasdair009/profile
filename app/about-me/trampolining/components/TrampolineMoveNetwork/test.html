<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
