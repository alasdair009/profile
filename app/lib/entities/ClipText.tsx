"use client";

import styled from "styled-components";
import {rem} from "polished";

export const ClipText = styled.span`
  align-items: center;
  //background: black;
  display: flex;
  font-size: ${rem(56)};
  height: 100%;
  justify-content: center;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  -webkit-text-fill-color: transparent;
  text-fill-color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  background: black;
  color: white;
`;