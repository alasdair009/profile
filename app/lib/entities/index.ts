"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
