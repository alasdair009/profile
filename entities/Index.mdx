import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
