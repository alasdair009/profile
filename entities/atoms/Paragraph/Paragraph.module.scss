@use "../../../app/styles/paragraph";

.root {
  @include paragraph.paragraphStyles;
}
