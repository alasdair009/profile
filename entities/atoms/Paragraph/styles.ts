import styled from "styled-components";
import {rem} from "polished";
import {colors} from "../../design-tokens/colors";
import {fonts, fontSizes, FontSizes} from "../../design-tokens/typography";
import {globalTextMaxWidth} from "../../design-tokens/dimensions";
import {Property} from "csstype";

export const Root = styled.p<{$fontSize: FontSizes, $align: Property.TextAlign}>`
  color: ${colors.whiteGhost};
  display: block;
  font-family: ${fonts.body};
  font-size: ${({$fontSize}) => fontSizes[$fontSize].rem};
  margin: 0 auto;
  max-width: ${rem(globalTextMaxWidth)};
  text-align: ${({$align}) => $align};
  width: 100%;
`;