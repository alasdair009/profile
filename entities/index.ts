export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
