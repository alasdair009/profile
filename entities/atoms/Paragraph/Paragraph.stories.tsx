import { Paragraph } from "./Paragraph";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";
import { colors, fontSizes } from "@/entities";

const meta: Meta<typeof Paragraph> = {
  component: Paragraph,
  argTypes: {
    children: {
      control: {
        type: "text",
      },
    },
    color: {
      control: {
        type: "color",
        presetColors: Object.values(colors),
      },
    },
  },
  args: {
    children: "Paragraph of text",
    align: "left",
    color: colors.whiteGhost,
    fontSize: "medium",
    textWrap: "wrap",
  },
};
export default meta;

export const Default: StoryObj<typeof Paragraph> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const paragraphElement = canvas.getByTestId(Paragraph.name);

    await expect(paragraphElement).toBeInTheDocument();
    await expect(paragraphElement).toHaveTextContent(`${args.children}`);
    await expect(paragraphElement).toHaveStyle(`text-align: ${args.align}`);
    await expect(paragraphElement).toHaveStyle(`color: ${args.color}`);
    await expect(paragraphElement).toHaveStyle(
      `font-size: ${fontSizes[args.fontSize ? args.fontSize : "medium"].px}`
    );
    await expect(paragraphElement).toHaveStyle(`text-wrap: ${args.textWrap}`);
  },
};

export const Clamped: StoryObj<typeof Paragraph> = {
  args: {
    children:
      "A very long line of text to demonstrate the text wrapping feature will cut the text off at the second wrap point. A very long line of text to demonstrate the text wrapping feature will cut the text off at the second wrap point. A very long line of text to demonstrate the text wrapping feature will cut the text off at the second wrap point.",
    lines: 2,
  },
};
