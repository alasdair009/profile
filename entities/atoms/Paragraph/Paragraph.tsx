import { Root } from "./styles";
import { HTMLAttributes, ReactNode } from "react";
import {FontSizes, TextAlignment} from "@/entities";

type ParagraphProps = {
  align?: TextAlignment;
  fontSize?: FontSizes;
  children: ReactNode;
} & HTMLAttributes<HTMLParagraphElement>;
export function Paragraph({
  align = "left",
  fontSize = "medium",
  children,
  ...rest
}: ParagraphProps) {
  return (
    <Root $align={align} $fontSize={fontSize} {...rest}>
      {children}
    </Root>
  );
}
