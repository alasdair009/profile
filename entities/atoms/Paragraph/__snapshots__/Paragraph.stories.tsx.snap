// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Paragraph Default play-test 1`] = `
<p data-testid="Paragraph"
   class="sc-hDcvty hzQlZe"
>
  Paragraph of text
</p>
`;
